app/domain/authentication/common/authenticate_with_resource_restrictions.rb

```diff
@@ -0,0 +1,50 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    AuthenticateWithResourceRestrictions = CommandClass.new(
```

Comment:
i don't like the naming very much. we still only validate the resource restrictions here so i would guess the class name should be `ValidateResourceRestrictions`.

I would also change the current `ValidateResourceRestrictions` to `ValidateResourceRestrictionsConfiguration` as that is what it does.